extensions/data-plane/data-plane-google-storage/src/test/java/org/eclipse/edc/connector/dataplane/gcp/storage/GcsDataSinkFactoryTest.java

```diff
@@ -32,51 +38,64 @@
 import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 class GcsDataSinkFactoryTest {
-
-    private final GcsDataSinkFactory factory = new GcsDataSinkFactory(
-            mock(ExecutorService.class),
-            mock(Monitor.class),
-            mock(Vault.class),
-            new JacksonTypeManager()
-    );
+    private final TypeManager typeManager = new JacksonTypeManager();
+    private Vault vault;
+    private IamService iamService;
+    private GcsDataSinkFactory factory;
+
+    @BeforeEach
+    void instantiateMocksAndVault() {
+        vault = mock();
+        iamService = mock();
+        factory = new GcsDataSinkFactory(
+                mock(ExecutorService.class),
+                mock(Monitor.class),
+                vault,
+                typeManager,
+                iamService
+        );
```

Comment:
nit: these assignments can be moved to the field declarations

Reply:
Done